src/coreclr/jit/fgopt.cpp

```diff
@@ -6400,10 +6400,9 @@ bool Compiler::fgUpdateFlowGraph(bool doTailDuplication, bool isPhase)
                 goto REPEAT;
             }
 
-            /* Remove unreachable or empty blocks - do not consider blocks marked BBF_DONT_REMOVE or genReturnBB block
-             * These include first and last block of a TRY, exception handlers and RANGE_CHECK_FAIL THROW blocks */
-
-            if ((block->bbFlags & BBF_DONT_REMOVE) == BBF_DONT_REMOVE || block == genReturnBB)
```

Comment:
Is the `genReturnBB` check just unnecessary? (obsolete?)

Reply:
Unnecessary; the block is [already always marked](https://github.com/dotnet/runtime/blob/946c5245aea149d9ece53fd0debc18328c29083b/src/coreclr/jit/flowgraph.cpp#L2357-L2359) with `BBF_DONT_REMOVE`.